sympy/solvers/solvers.py

```diff
@@ -948,27 +1021,23 @@ def _sympified_list(w):
                 else Abs(arg))
             if was == fi:
                 break
-
         for e in fi.find(Abs):
             if e.has(*symbols):
                 raise NotImplementedError('solving %s when the argument '
                     'is not real or imaginary.' % e)
-
         # arg
         fi = fi.replace(arg, lambda a: arg(a).rewrite(atan2).rewrite(atan))
-
         # save changes
         f[i] = fi
-
     # see if re(s) or im(s) appear
     freim = [fi for fi in f if fi.has(re, im)]
     if freim:
         irf = []
         for s in symbols:
-            if s.is_real or s.is_imaginary:
+            if s.is_extended_real or s.is_imaginary:
                 continue  # neither re(x) nor im(x) will appear
             # if re(s) or im(s) appear, the auxiliary equation must be present
-            if any(fi.has(re(s), im(s)) for fi in freim):
+            if any(fi.has(re(s), im(s)) for fi in f):
```

Comment:
Why has this changed?

Reply:
That's a mistake on my part. The `f` was supposed to be `freim`.